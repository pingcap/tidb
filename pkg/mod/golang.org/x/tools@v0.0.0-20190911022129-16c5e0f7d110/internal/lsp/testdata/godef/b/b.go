package b

import (
	myFoo "golang.org/x/tools/internal/lsp/foo" //@mark(myFoo, "myFoo"),godef("foo", PackageFoo),godef("myFoo", myFoo)
	"golang.org/x/tools/internal/lsp/godef/a"   //@mark(AImport, "\"")
)

type S1 struct { //@S1
	F1  int //@mark(S1F1, "F1")
	S2      //@godef("S2", S2), mark(S1S2, "S2")
	a.A     //@godef("A", A)
}

type S2 struct { //@S2
	F1   string //@mark(S2F1, "F1")
	F2   int    //@mark(S2F2, "F2")
	*a.A        //@godef("A", A),godef("a",AImport)
}

type S3 struct {
	F1 struct {
		a.A //@godef("A", A)
	}
}

func Bar() {
	a.Stuff()   //@godef("Stuff", Stuff)
	var x S1    //@godef("S1", S1)
	_ = x.S2    //@godef("S2", S1S2)
	_ = x.F1    //@godef("F1", S1F1)
	_ = x.F2    //@godef("F2", S2F2)
	_ = x.S2.F1 //@godef("F1", S2F1)

	var _ *myFoo.StructFoo //@godef("myFoo", myFoo)
}
