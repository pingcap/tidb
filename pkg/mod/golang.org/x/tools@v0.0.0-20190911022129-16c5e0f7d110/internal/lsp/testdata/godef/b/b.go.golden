-- A-definition --
godef/a/a.go:7:6-7: defined here as A string //@A

-- A-definition-json --
{
	"span": {
		"uri": "file://godef/a/a.go",
		"start": {
			"line": 7,
			"column": 6,
			"offset": 75
		},
		"end": {
			"line": 7,
			"column": 7,
			"offset": 76
		}
	},
	"description": "A string //@A"
}

-- A-hover --
A string //@A

-- AImport-definition --
godef/b/b.go:5:2: defined here as package a ("golang.org/x/tools/internal/lsp/godef/a")

-- AImport-definition-json --
{
	"span": {
		"uri": "file://godef/b/b.go",
		"start": {
			"line": 5,
			"column": 2,
			"offset": 121
		},
		"end": {
			"line": 5,
			"column": 2,
			"offset": 121
		}
	},
	"description": "package a (\"golang.org/x/tools/internal/lsp/godef/a\")"
}

-- AImport-hover --
package a ("golang.org/x/tools/internal/lsp/godef/a")
-- PackageFoo-definition --
foo/foo.go:1:9-12: defined here as 
-- PackageFoo-definition-json --
{
	"span": {
		"uri": "file://foo/foo.go",
		"start": {
			"line": 1,
			"column": 9,
			"offset": 8
		},
		"end": {
			"line": 1,
			"column": 12,
			"offset": 11
		}
	},
	"description": ""
}

-- PackageFoo-hover --
myFoo "golang.org/x/tools/internal/lsp/foo" //@mark(myFoo, "myFoo"),godef("foo", PackageFoo),godef("myFoo", myFoo)

-- S1-definition --
godef/b/b.go:8:6-8: defined here as S1 struct {
	F1  int //@mark(S1F1, "F1")
	S2      //@godef("S2", S2), mark(S1S2, "S2")
	a.A     //@godef("A", A)
}

-- S1-definition-json --
{
	"span": {
		"uri": "file://godef/b/b.go",
		"start": {
			"line": 8,
			"column": 6,
			"offset": 196
		},
		"end": {
			"line": 8,
			"column": 8,
			"offset": 198
		}
	},
	"description": "S1 struct {\n\tF1  int //@mark(S1F1, \"F1\")\n\tS2      //@godef(\"S2\", S2), mark(S1S2, \"S2\")\n\ta.A     //@godef(\"A\", A)\n}"
}

-- S1-hover --
S1 struct {
	F1  int //@mark(S1F1, "F1")
	S2      //@godef("S2", S2), mark(S1S2, "S2")
	a.A     //@godef("A", A)
}
-- S1F1-definition --
godef/b/b.go:9:2-4: defined here as field F1 int

-- S1F1-definition-json --
{
	"span": {
		"uri": "file://godef/b/b.go",
		"start": {
			"line": 9,
			"column": 2,
			"offset": 215
		},
		"end": {
			"line": 9,
			"column": 4,
			"offset": 217
		}
	},
	"description": "field F1 int"
}

-- S1F1-hover --
@mark(S1F1, "F1")
field F1 int
-- S1S2-definition --
godef/b/b.go:10:2-4: defined here as field S2 S2

-- S1S2-definition-json --
{
	"span": {
		"uri": "file://godef/b/b.go",
		"start": {
			"line": 10,
			"column": 2,
			"offset": 244
		},
		"end": {
			"line": 10,
			"column": 4,
			"offset": 246
		}
	},
	"description": "field S2 S2"
}

-- S1S2-hover --
@godef("S2", S2), mark(S1S2, "S2")
field S2 S2
-- S2-definition --
godef/b/b.go:14:6-8: defined here as S2 struct {
	F1   string //@mark(S2F1, "F1")
	F2   int    //@mark(S2F2, "F2")
	*a.A        //@godef("A", A),godef("a",AImport)
}

-- S2-definition-json --
{
	"span": {
		"uri": "file://godef/b/b.go",
		"start": {
			"line": 14,
			"column": 6,
			"offset": 323
		},
		"end": {
			"line": 14,
			"column": 8,
			"offset": 325
		}
	},
	"description": "S2 struct {\n\tF1   string //@mark(S2F1, \"F1\")\n\tF2   int    //@mark(S2F2, \"F2\")\n\t*a.A        //@godef(\"A\", A),godef(\"a\",AImport)\n}"
}

-- S2-hover --
S2 struct {
	F1   string //@mark(S2F1, "F1")
	F2   int    //@mark(S2F2, "F2")
	*a.A        //@godef("A", A),godef("a",AImport)
}
-- S2F1-definition --
godef/b/b.go:15:2-4: defined here as field F1 string

-- S2F1-definition-json --
{
	"span": {
		"uri": "file://godef/b/b.go",
		"start": {
			"line": 15,
			"column": 2,
			"offset": 342
		},
		"end": {
			"line": 15,
			"column": 4,
			"offset": 344
		}
	},
	"description": "field F1 string"
}

-- S2F1-hover --
@mark(S2F1, "F1")
field F1 string
-- S2F2-definition --
godef/b/b.go:16:2-4: defined here as field F2 int

-- S2F2-definition-json --
{
	"span": {
		"uri": "file://godef/b/b.go",
		"start": {
			"line": 16,
			"column": 2,
			"offset": 375
		},
		"end": {
			"line": 16,
			"column": 4,
			"offset": 377
		}
	},
	"description": "field F2 int"
}

-- S2F2-hover --
@mark(S2F2, "F2")
field F2 int
-- Stuff-definition --
godef/a/a.go:9:6-11: defined here as func a.Stuff()

-- Stuff-definition-json --
{
	"span": {
		"uri": "file://godef/a/a.go",
		"start": {
			"line": 9,
			"column": 6,
			"offset": 95
		},
		"end": {
			"line": 9,
			"column": 11,
			"offset": 100
		}
	},
	"description": "func a.Stuff()"
}

-- Stuff-hover --
func a.Stuff()
-- myFoo-hover --
package myFoo ("golang.org/x/tools/internal/lsp/foo")
